guides/ssb-server-context.md

```diff
@@ -0,0 +1,36 @@
+# ssb-server, ssb-client & Core Functions
+
+## What is ssb-server?
+[ssb-server](https://github.com/ssbc/ssb-server) is the component of secure-scuttlebutt that performs the peer-to-peer connections, and reading and writing to/from the append-only log. It can also be extended with plugins. It exposes via its API core functionality such as reading from, and writing to, the append-only log, as well as the extended functions provided by plugins utilized. This process should be thought of as a sort of "background" process, since it can only perform the peer-to-peer "gossiping" functions if it is running. An ssb-server is one to one coupled with an "identity" which is essentially the secret key associated with a given append-only log.
+
+> Sometimes, ssb-server is packaged as an "embedded" process with an SSB application, such as Patchwork. Two or more versions of ssb-server connecting to the same identity and feed (log) should not run at the same time. The SSB ecosystem is currently exploring a variety of solutions to this problem.
+
+There are a variety of ways to access the API functions of ssb-server.
+
+If the ssb-server process is started using the ssb-server command line command, it can be accessed:
+- via a separate/second command line interface, using other ssb-server command line commands
+- via ssb-client in a nodejs program, which connects to it via RPC (remote procedure calls) and has complete access to the API including extended functionality from plugins
+
+If the ssb-server process is started programmatically, by importing it into a nodejs file, it can be accessed:
+- directly via the in-program reference to the running server
+- via a separate/second command line interface, using other ssb-server command line commands
+- via ssb-client in a nodejs program, which connects to it via RPC (remote procedure calls) and has complete access to the API including extended functionality from plugins
```

Comment:
gotcha : it _can_ have complete access if and only if `ssb-master` is installed as a plugin. (I think)

Reply:
@mixmix I can't quite make sense of this... https://github.com/ssbc/ssb-master/blob/master/index.js

Reply:
what is the access it has if you DON'T include ssb-master plugin?

Reply:
Now reading the test suite, and thinking...

if you call `use(require('ssb-master'))`, then you CAN pass a `master` prop into `createSsbServer`, and that creates a restriction on calls to it...
https://github.com/ssbc/ssb-master/blob/master/test/master.js#L22